src/lib/eet/eet_cipher.c

```diff
@@ -190,36 +190,33 @@ eet_identity_open(const char               *certificate_file,
 
 # else /* ifdef HAVE_GNUTLS */
    /* Openssl private declarations */
-   FILE *fp;
    EVP_PKEY *pkey = NULL;
    X509 *cert = NULL;
 
    if (!emile_cipher_init()) return NULL;
 
    /* Load the X509 certificate in memory. */
-   fp = fopen(certificate_file, "rb");
-   if (!fp)
-     return NULL;
-
-   cert = PEM_read_X509(fp, NULL, NULL, NULL);
-   fclose(fp);
-   if (!cert)
-     goto on_error;
+   {
+      BIO* cert_bio = BIO_new_file(certificate_file, "rb");
+      cert = PEM_read_bio_X509(cert_bio, NULL, NULL, NULL);
+      BIO_free(cert_bio);
+      if (!cert)
+        goto on_error;
+   }
```

Comment:
do you know if this will work too for cygwin?

Reply:
maybe we could guard to make this way only for `_MSC_VER`?

Reply:
> do you know if this will work too for cygwin?

Of course it will, this is OpenSSL's API.

> maybe we could guard to make this way only for `_MSC_VER`?

There's no need for it, this isn't a platform-dependent code in any form.